aspnetcore/mvc/advanced/app-parts/3.0sample1/WebAppParts/StartupViews.cs

```diff
@@ -0,0 +1,65 @@
+﻿using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.Hosting;
+using MySharedApp.Controllers;
+using System.Reflection;
+
+namespace WebAppParts
+{
+    public class StartupViews
+    {
+        public StartupViews(IConfiguration configuration)
+        {
+            Configuration = configuration;
+        }
+
+        public IConfiguration Configuration { get; }
+
+        #region snippet
+        // requires 
+        // using Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation;
+        // using System.Reflection;
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.Configure<MvcRazorRuntimeCompilationOptions>(options =>
+            {
+                options.FileProviders.Add(
+                 new EmbeddedFileProvider(typeof(MySharedController).GetTypeInfo().Assembly));
```

Comment:
Let's not document this. RCLs should be a sufficient replacement for this (which the first sample covers).